calabash-cucumber/lib/calabash-cucumber/map.rb

```diff
@@ -143,6 +144,40 @@ def self.assert_map_results(map_results, msg)
         end
       end
 
+      # Evaluating whether a query contain correct predicate selector
+      # returns true if query include correct predicate selector
+      def self.correct_predicate?(query)
+        if !query.match(/{.*}/).nil?
```

Comment:
There are actually two conditions for a valid predicate:

1. The predicate keyword is supported (CONTAINS, LIKE, etc).
2. The right-hand side of the predicate is valid (not missing a `'`).